Tests/MoyaProviderSpec.swift

```diff
@@ -973,5 +972,63 @@ final class MoyaProviderSpec: QuickSpec {
                 }
             }
         }
+
+        describe("a provider for stubbed requests with validation") {
+            var stubbedProvider: MoyaProvider<GitHub>!
+
+            context("response contains invalid status code") {
+                it("returns a failure case") {
+                    let endpointClosure = { (target: GitHub) -> Endpoint in
+                        return Endpoint(
+                            url: URL(target: target).absoluteString,
+                            sampleResponseClosure: { .networkResponse(400, target.sampleData) },
+                            method: target.method,
+                            task: target.task,
+                            httpHeaderFields: target.headers
+                        )
+                    }
+                    stubbedProvider = MoyaProvider<GitHub>(endpointClosure: endpointClosure, stubClosure: MoyaProvider.immediatelyStub)
+                    stubbedProvider.request(.zen) { result in
+                        let validStatusCodes = GitHub.zen.validationType.statusCodes
+                        switch result {
+                        case .success(let response):
+                            expect(validStatusCodes).to(contain(response.statusCode))
+                        case .failure(let error):
+                            guard case let MoyaError.underlying(_, response) = error else {
+                                fatalError("Expected MoyaError.underlying for validation failure.")
+                            }
+                            expect(validStatusCodes).toNot(contain(response!.statusCode))
+                        }
+                    }
+                }
+            }
+            context("response contains valid status code") {
+                it("returns a success case") {
+                    let endpointClosure = { (target: GitHub) -> Endpoint in
+                        return Endpoint(
+                            url: URL(target: target).absoluteString,
+                            sampleResponseClosure: { .networkResponse(200, target.sampleData) },
+                            method: target.method,
+                            task: target.task,
+                            httpHeaderFields: target.headers
+                        )
+                    }
+                    stubbedProvider = MoyaProvider<GitHub>(endpointClosure: endpointClosure, stubClosure: MoyaProvider.immediatelyStub)
+                    let validStatusCodes = GitHub.zen.validationType.statusCodes
+                    stubbedProvider.request(.zen) { result in
+                        switch result {
+                        case .success(let response):
+                            expect(validStatusCodes).toNot(beEmpty())
+                            expect(validStatusCodes).to(contain(response.statusCode))
+                        case .failure(let error):
+                            guard case let MoyaError.underlying(_, response) = error else {
```

Comment:
here a `failure` case is probably a failed test, as we expect only `success` case, right? and similar in the above test with the `success case`?